packages/react-tooltip/src/components/Tooltip/useTooltip.tsx

```diff
@@ -38,50 +29,50 @@ const arrowHeight = 6; // Update the arrow's width/height in useTooltipStyles.ts
  *
  * {@docCategory Tooltip}
  */
-export const useTooltip = (
-  props: TooltipProps,
-  ref: React.Ref<HTMLElement>,
-  defaultProps?: TooltipProps,
-): TooltipState => {
+export const useTooltip = (props: TooltipProps, ref: React.Ref<HTMLElement>): TooltipState => {
   const context = React.useContext(TooltipContext);
   const isServerSideRender = useIsSSR();
   const { targetDocument } = useFluent();
   const [setDelayTimeout, clearDelayTimeout] = useTimeout();
 
-  const state = mergeProps(
-    {
-      ref,
-      children: props.children,
-      content: {
-        as: React.Fragment,
-      },
-      id: useId('tooltip-'),
-      role: 'tooltip',
-      showDelay: 250,
-      hideDelay: 250,
-      triggerAriaAttribute: 'label',
-    },
-    defaultProps && resolveShorthandProps(defaultProps, tooltipShorthandProps),
-    resolveShorthandProps(props, tooltipShorthandProps),
-  );
-
-  const [visible, setVisibleInternal] = useControllableState({ state: state.visible, initialState: false });
+  const { onVisibleChange } = props;
+  const [visible, setVisibleInternal] = useControllableState({ state: props.visible, initialState: false });
   const setVisible = React.useCallback(
     (newVisible: boolean, ev?: React.PointerEvent<HTMLElement> | React.FocusEvent<HTMLElement>) => {
       clearDelayTimeout();
       setVisibleInternal(oldVisible => {
         if (newVisible !== oldVisible) {
-          const onVisibleChange = state.onVisibleChange; // Workaround for bug in react-exhaustive-deps lint rule
           onVisibleChange?.(ev, { visible: newVisible });
         }
         return newVisible;
       });
     },
-    [clearDelayTimeout, setVisibleInternal, state.onVisibleChange],
+    [clearDelayTimeout, setVisibleInternal, onVisibleChange],
   );
 
-  state.visible = visible;
-  state.shouldRenderTooltip = visible;
+  const state: TooltipState = {
+    // Default values
+    id: useId('tooltip-', props.id),
+    role: 'tooltip',
+    showDelay: 250,
+    hideDelay: 250,
+    triggerAriaAttribute: 'label',
+    shouldRenderTooltip: visible,
+
+    // User-provided props
+    ...props,
+    ref,
+
+    // Controlled state values
+    visible,
+
+    // Slots
+    components: {
+      root: 'div',
+      content: React.Fragment,
```

Comment:
According to `TooltipSlots` this should be `div` or another HTML element:

```tsx
content: React.HTMLAttributes<HTMLElement>;
```

---

We should modify `TooltipSlots` or an element there as code below will a produce a warning (and does not has sense):

```tsx
<Tooltip content={{ id: 'foo' }} /> // HTML attributes cannot be applied to fragments
```

Reply:
`content` was never a real slot; it's more like the children of the tooltip container. I changed it to a prop of type React.ReactNode instead.